### ISSUE TITLE
Python 3.8 Backward Compatibility due to `ast.unparse`

### ISSUE THREAD

**Issue by zhongxiang117** [mdtraj/mdtraj#1905]
I downloaded least `mdtraj` and try to compile and run it by using `python 3.8`, however, I found a problem saying missing attribute in `ast.unparse` in this line: [https://github.com/mdtraj/mdtraj/blob/master/mdtraj/core/selection.py#L27](https://github.com/mdtraj/mdtraj/blob/master/mdtraj/core/selection.py#L27).

I carefully checked the commit history and issue reports, found #1613 and #1551, and made sure that this broke change happens at [https://github.com/mdtraj/mdtraj/commit/d3d9891a66fd5b951fcbae455c4cb76ac4d1a004](https://github.com/mdtraj/mdtraj/commit/d3d9891a66fd5b951fcbae455c4cb76ac4d1a004).

In a nutshell, method `ast.unparse` is added to python from `version 3.9`, (refer to [here](https://docs.python.org/3/library/ast.html#ast.unparse)). Thus, to fix it, either restore the broken change, or classify the python version in [https://github.com/mdtraj/mdtraj/blob/master/setup.py#L49](https://github.com/mdtraj/mdtraj/blob/master/setup.py#L49)

**Comment by sukritsingh** [comment]
Hi! Thanks for pointing this out. For the moment we're no longer really supporting python 3.8 (and we don't even include it in our CI tests anymore for). Is there any particular reason you need to use python `3.8`?

If nothing else, perhaps its worth version constraining ourselves too....

**Comment by zhongxiang117** [comment]
OK, thanks for your reply. No, no particular reason, just simply keep the consistent of my early projects.

I saw a version requirement of this line [https://github.com/mdtraj/mdtraj/blob/master/setup.py#L305](https://github.com/mdtraj/mdtraj/blob/master/setup.py#L305). It's time for me to use the newer version of python.

By the way, when I tried to compile it from source codes, failed many times on the complaining "undefined symbol". Carefully checking codes, I also saw the requirement of [https://github.com/mdtraj/mdtraj/blob/master/pyproject.toml#L8](https://github.com/mdtraj/mdtraj/blob/master/pyproject.toml#L8), and a discussion in [https://github.com/mdtraj/mdtraj/discussions/1793](https://github.com/mdtraj/mdtraj/discussions/1793), no offense, I mean all those version requirements should be stated clearly, (they are clear, but not that much clear), for example, in the doc, in readme, or in one conclusion file. Thanks for your considerations.

I am going to close this issue. Thanks you all for your work.

**Comment by sukritsingh** [comment]
Hopefully your installation on the newer version of python goes smoothly! 

> I mean all those version requirements should be stated clearly, (they are clear, but not that much clear), for example, in the doc, in readme, or in one conclusion file.

Thanks for this feedback. Yes, we are aware that the docs are in sore need of an update (contributions welcome!). It's an ongoing effort as we get the repo back up and smoothly running.
